Add english docs for UIE (#4255)
PaddlePaddle/PaddleNLP: applications/information_extraction/taskflow_doc_en.md

# UIE Taskflow User Guide

> **Table of contents**
> - [1. Function Introduction](#1)
> - [2. Document Information Extraction](#2)
>    - [2.1 Entity Extraction](#21)
>    - [2.2 Relation Extraction](#22)
>    - [2.3 Use across tasks](#23)
>    - [2.4 Input Description](#24)
>    - [2.5 Tips](#25)
>    - [2.6 Results Visualization](#26)
>    - [2.7 More Configuration](#27)

Reviewer: ```suggestion
**Table of contents**
- [1. Introduction](#1)
- [2. Document Information Extraction](#2)
   - [2.1 Entity Extraction](#21)
   - [2.2 Relation Extraction](#22)
   - [2.3 Multi-Task Extraction](#23)
   - [2.4 Input Format](#24)
   - [2.5 Tips](#25)
   - [2.6 Visualization](#26)
   - [2.7 More Configuration](#27)
```

<a name="1"></a>

> ## 1. Function Introduction
> 
> ```paddlenlp.Taskflow``` provides general information extraction of text and documents, evaluation opinion extraction and other capabilities, and can extract various types of information, including but not limited to named entity recognition (such as person name, place name, organization name, etc.), Relationships (such as the director of the movie, the release time of the song, etc.), events (such as a car accident at a certain intersection, an earthquake in a certain place, etc.), and information such as evaluation dimensions, opinion words, and emotional tendencies. Users can use natural language to customize the extraction target, and can uniformly extract the corresponding information in the input text or document without training. ** Realize out-of-the-box use and meet various information extraction needs **

Reviewer: ```suggestion
## 1. Introduction

```paddlenlp.Taskflow``` provides general information extraction of text and documents, evaluation opinion extraction and other capabilities, and can extract various types of information, including but not limited to named entities (such as person name, place name, organization name, etc.), relations (such as the director of the movie, the release time of the song, etc.), events (such as a car accident at a certain intersection, an earthquake in a certain place, etc.), and information such as product reviews, opinions, and sentiments. Users can use natural language to customize the extraction target, and can uniformly extract the corresponding information in the input text or document without training.
```

> ## 1. Function Introduction
> 
> ```paddlenlp.Taskflow``` provides general information extraction of text and documents, evaluation opinion extraction and other capabilities, and can extract various types of information, including but not limited to named entity recognition (such as person name, place name, organization name, etc.), Relationships (such as the director of the movie, the release time of the song, etc.), events (such as a car accident at a certain intersection, an earthquake in a certain place, etc.), and information such as evaluation dimensions, opinion words, and emotional tendencies. Users can use natural language to customize the extraction target, and can uniformly extract the corresponding information in the input text or document without training. ** Realize out-of-the-box use and meet various information extraction needs **

Reviewer: ```suggestion
## 1. Introduction

```paddlenlp.Taskflow``` provides general information extraction of text and documents, evaluation opinion extraction and other capabilities, and can extract various types of information, including but not limited to named entities (such as person name, place name, organization name, etc.), relations (such as the director of the movie, the release time of the song, etc.), events (such as a car accident at a certain intersection, an earthquake in a certain place, etc.), and information such as product reviews, opinions, and sentiments. Users can use natural language to customize the extraction target, and can uniformly extract the corresponding information in the input text or document without training.
```

<a name="2"></a>

## 2. Document Information Extraction

> This chapter mainly introduces the document extraction function of Taskflow, the following example picture [download link](https://bj.bcebos.com/paddlenlp/taskflow/information_extraction/cases.zip).

Reviewer: ```suggestion
This section introduces the document extraction capability of Taskflow with the following example picture [download link](https://bj.bcebos.com/paddlenlp/taskflow/information_extraction/cases.zip).
```

<a name="21"></a>

#### 2.1 Entity Extraction

> Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. In the open domain information extraction, the extracted categories are not limited, and users can define them by themselves.

Reviewer: ```suggestion
Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. UIE adopts the open-domain approach where the entity category is not fixed and the users can define them by through natural language.
```

> Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. In the open domain information extraction, the extracted categories are not limited, and users can define them by themselves.

Reviewer: ```suggestion
Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. UIE adopts the open-domain approach where the entity category is not fixed and the users can define them by through natural language.
```

> Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. In the open domain information extraction, the extracted categories are not limited, and users can define them by themselves.

Reviewer: ```suggestion
Entity extraction, also known as Named Entity Recognition (NER for short), refers to identifying entities with specific meanings in text. UIE adopts the open-domain approach where the entity category is not fixed and the users can define them by through natural language.
```

> - Customs declaration

Reviewer: ```suggestion
- Example: Customs Declaration Form
```

            'probability': 0.9950633161231508,
            'start': 238,
            'text': '一般征税'}],
'收发货人': [{'bbox': [[321, 533, 841, 580]],
            'end': 95,
            'probability': 0.4772132061042136,
            'start': 82,
            'text': '上海新尚实国际贸易有限公司'},
        {'bbox': [[306, 584, 516, 624]],
            'end': 150,
            'probability': 0.33807074572195006,
            'start': 140,
            'text': '31222609K9'}],
'运输工具名称': [{'bbox': [[1306, 672, 1516, 712], [1549, 668, 1645, 712]],
            'end': 190,
            'probability': 0.6692050414718089,
            'start': 174,
            'text': 'E. R. TIANAN004E'}],
'运输方式': [{'bbox': [[1070, 664, 1240, 715]],
            'end': 174,
            'probability': 0.9994416347044179,
            'start': 170,
            'text': '永路运输'}],
'进口口岸': [{'bbox': [[1070, 566, 1346, 617]],
            'end': 120,
            'probability': 0.9945697196994345,
            'start': 111,
            'text': '洋山港区-2248'}],
'进口日期': [{'bbox': [[1726, 569, 1933, 610]],
            'end': 130,
            'probability': 0.9804819494073627,
            'start': 120,
            'text': '2017-02-24'}]}]
```

> - ID

Reviewer: ```suggestion
- Example: Driver's License
```

> - ID

Reviewer: ```suggestion
- Example: Driver's License
```

<div align="center">
    <img src=https://user-images.githubusercontent.com/40840292/206114081-8c82e2a2-0c88-4ca3-9651-b12c94266be9.png height=400 width=700 hspace='10'/>
</div>

```python
>>> from pprint import pprint
>>> from paddlenlp import Taskflow
>>> schema = ["Name", "Date of birth", "Issue date"]
>>> ie = Taskflow("information_extraction", schema=schema, model="uie-x-base", ocr_lang="en", schema_lang="en")
>>> pprint(ie({"doc": "./cases/license.jpeg"}))
```

<a name="22"></a>

#### 2.2 Relation Extraction

Relation Extraction refers to identifying entities from text and extracting the semantic relationship between entities, and then obtaining triple information, namely <subject, predicate, object>.

> - Sheet

Reviewer: ```suggestion
- Example: Extracting relations from a table
```

<div align="center">
    <img src=https://user-images.githubusercontent.com/40840292/206115688-30de315a-8fd4-4125-a3c3-8cb05c6e39e5.png height=180 width=600 hspace='10'/>
</div>

```python
>>> from pprint import pprint
>>> from paddlenlp import Taskflow
>>> schema = {"姓名": ["招聘单位", "报考岗位"]}
>>>> ie = Taskflow("information_extraction", schema=schema, model="uie-x-base")
>>> pprint(ie({"doc": "./cases/table.png"}))
```

<a name="23"></a>

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

> #### 2.3 Use across tasks
> 
> - Multi-task extraction
> 
> To extract entities and relation from documents, the schema structure is as follows:

Reviewer: ```suggestion
#### 2.3 Multi-Task Extraction

To extract entities and relation from documents simultaneously, you may set the schema structure as following:
```

```text
schema = [
    "Total GBP",
    "No.",
    "Date",
    "Customer No.",
    "Subtotal without VAT",
    {
        "Description": [
            "Quantity",
            "Amount"
        ]
    }
]
```

<div align="center">
    <img src=https://user-images.githubusercontent.com/40840292/206120861-13b475dc-9a78-43bc-9dec-91f331db2ddf.png height=400 width=650 hspace='10'/>
</div>

```python
>>> from pprint import pprint
>>> from paddlenlp import Taskflow

>>> schema = ["Total GBP", "No.", "Date", "Customer No.", "Subtotal without VAT", {"Description": ["Quantity", "Amount"]}]
>>> ie = Taskflow("information_extraction", schema=schema, model="uie-x-base", ocr_lang="en", schema_lang="en")
>>> pprint(ie({"doc": "./cases/delivery_note.png"}))
```

<a name="24"></a>

#### 2.4 Input Description

- input format

> For document information extraction, UIE-X supports image paths, http image links, and base64 input forms, and supports image and PDF document formats. In a dict `text` indicates text input and `doc` refer to the document input.

Reviewer: ```suggestion
#### 2.4 Input Format

For document information extraction, UIE-X supports image paths, http image links, base64 input form, and image and PDF document formats. In the input dict, `text` indicates text input and `doc` refer to the document input.
```

<div align="center">
    <img src=https://user-images.githubusercontent.com/40840292/206139057-aedec98f-683c-4648-999d-81ce5ea04a86.png height=250 width=500 hspace='10'/>
</div>

```python
>>> from pprint import pprint
>>> from paddlenlp import Taskflow

>>> schema = "中标候选人名称"
>>> ie = Taskflow("information_extraction", schema=schema, model="uie-x-base", layout_analysis=True)
>>> pprint(ie({"doc": "https://gimg2.baidu.com/image_search/src=http%3A%2F%2Fwww.xuyiwater.com%2Fwp-content%2Fuploads%2F2021%2F06%2F1-4.jpg&refer=http%3A%2F%2Fwww.xuyiwater.com&app=2002&size=f9999,10000&q=a80&n=0&g=0n&fmt=auto?sec=1672994926&t=2a4a3fedf6999a34ccde190f97bcfa47"}))
```

<div align="center">
    <img src=https://user-images.githubusercontent.com/40840292/206137978-3a69e7e2-dc2e-4d11-98b7-25911b0375a0.png height=350 width=600 hspace='10'/>
</div>

```python
>>> schema = "抗血小板药物的用药指征"
>>> ie.set_schema(schema)
>>> pprint(ie({"doc": "./cases/drug.webp"}))
```

<a name="26"></a>

> #### 2.6 Results Visualization

Reviewer: ```suggestion
#### 2.6 Visualization
```
